extensions/platform-http/runtime/src/main/java/org/apache/camel/quarkus/component/platform/http/runtime/PlatformHttpRecorder.java

```diff
@@ -37,7 +39,20 @@
     }
 
     public RuntimeValue<PlatformHttpComponent> createComponent(RuntimeValue<PlatformHttpEngine> engine) {
-        PlatformHttpComponent component = new PlatformHttpComponent();
+        PlatformHttpComponent component = new PlatformHttpComponent() {
+            @Override
+            protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
+                // remove matchOnUriPrefix as it will be fixed by camel 3.2 but will cause the context
+                // to fail as the property cannot be bound to the enpoint.
+                //
+                // TODO: remove once migrating to camel 3.2
+                parameters.remove("matchOnUriPrefix");
```

Comment:
Shouldn't we warn at least that matchOnUriPrefix is ignored?

Reply:
we could but actually the property is only added by platform-http component itself (when generating rest endpoints) even if it is not a property it  is able to handles so basically it would warn itself